src/api-server-selector.js

```diff
@@ -197,6 +221,32 @@ export class ApiServerSelector extends EventsTargetMixin(AmfHelperMixin(LitEleme
     node.removeEventListener('api-navigation-selection-changed', this._handleNavigationChange);
   }
 
+  _isValueValid(value) {
+    if (this.isCustom) {
+      return true;
+    }
+
+    return Boolean(this._findServerByValue(value));
+  }
+
+  _findServerByValue(value) {
+    const { servers = [] } = this;
+    return servers.find(server => this._getServerUri(server) === value)
+  }
+
+  _getIndexForValue(value) {
+    if (this.isCustom) {
+      return this._getCustomUriIndex();
+    }
+
+    if (this.selectedType === 'slot') {
+      return this._getCustomUriIndex() - 1;
```

Comment:
We can't do this as we don't actually now if the client is going to attach one or more than one elements to the slot. The slot may contain n custom base uris. We should instead search for the attached Elements, see if any of them matches to the value, and return the correct index of the slot in the list. I can take this change if you want